scipy/stats/stats.py

```diff
@@ -6636,32 +6636,29 @@ def _count_paths_outside_method(m, n, g, h):
     xj = [(h + mg * j + ng-1)//ng for j in range(lxj)]
     # B is an array just holding a few values of B(x,y), the ones needed.
     # B[j] == B(x_j, j)
-    if lxj == 0:
-        return np.round(special.binom(m + n, n))
-    B = np.zeros(lxj)
-    B[0] = 1
-    # Compute the B(x, y) terms
-    # The binomial coefficient is an integer, but special.binom() may return a float.
-    # Round it to the nearest integer.
-    for j in range(1, lxj):
-        Bj = np.round(special.binom(xj[j] + j, j))
-        if not np.isfinite(Bj):
-            raise FloatingPointError()
-        for i in range(j):
-            bin = np.round(special.binom(xj[j] - xj[i] + j - i, j-i))
-            Bj -= bin * B[i]
-        B[j] = Bj
-        if not np.isfinite(Bj):
-            raise FloatingPointError()
-    # Compute the number of path extensions...
-    num_paths = 0
-    for j in range(lxj):
-        bin = np.round(special.binom((m-xj[j]) + (n - j), n-j))
-        term = B[j] * bin
-        if not np.isfinite(term):
-            raise FloatingPointError()
-        num_paths += term
-    return np.round(num_paths)
+    try:
+        if lxj == 0:
+            return special.comb(m + n, n, exact=True)
+        B = np.zeros(lxj)
+        B[0] = 1
+        # Compute the B(x, y) terms
+        # The binomial coefficient is an integer, so special.comb(,exact=True)
+        # can be used instead of special.binom()
+        for j in range(1, lxj):
+            Bj = special.comb(xj[j] + j, j,exact=True)
+            for i in range(j):
+                bin = special.comb(xj[j] - xj[i] + j - i, j-i,exact=True)
+                Bj -= bin * B[i]
+            B[j] = Bj
+        # Compute the number of path extensions...
+        num_paths = 0
+        for j in range(lxj):
+            bin = special.comb((m-xj[j]) + (n - j), n-j,exact=True)
+            term = B[j] * bin
+            num_paths += term
+        return num_paths
+    except (RuntimeWarning,OverflowError):
```

Comment:
catching a warning and turning it into an error is a bit odd. A `RuntimeWarning` could be caused by a lot of things, so this doesn't look quite right. What is the exact cause of the warning you're trying to catch, and can you deal with that another way?

Reply:
If it's the `if not np.isfinite(term):`, maybe a more granular method would be using `with np.errstate(invalid='raise'):`?

Reply:
Yeah, I thought that might be problematic, but I did not think of that solution. I try to catch the runtime warning that comes from the overflow. The outer function catches a FloatingPointError which was raised before by checking np.isfinite. The whole error handling could probably improved by using np.errstate in the outer function as well.

Reply:
Should it be `np.errstate(invalid='raise')` or `np.errstate(over='raise')`? (The FloatingPointError was only raised explicitly if the computation overflowed.)
Does this catch overflows which occur in `scipy.special.binom` or only Python code under the `np.errstate` context?

Reply:
It does not seem to catch overflows in special.binom. `np.errstate(invalid='raise')` produces errors instead of warnings, as can be seen in the test cases, so I think it should be correct.

Reply:
The computation involving binomials doesn't have to deal with invalid operations, only overflows arising from binomial coefficients being larger than `np.info(float).max` (or multiplication of such binomials by `B[j]`).  E.g. `scipy.special.binom(1100, 512)` returns `np.inf`.  If the j-th term is infinite, the computation bails out, returning FloatingPointError(). OverflowError() may have also been a possible Exception to use for this situation.
So the context should be:
  `with np.errstate(over='raise'):`